### PR TITLE
add item() to all backends

### PR DESCRIPTION
**Pull request by mganahl**

### CODE REVIEW

File: tensornetwork/backends/tensorflow/tensorflow_backend.py [hunk 1]
```diff
@@ -383,3 +385,15 @@ def sign(self, tensor: Tensor) -> Tensor:
       tensor: The input tensor.
     """
     return tf.math.sign(tensor)
+
+  def item(self, tensor):
+    numel = 0
+    if len(tensor.shape) > 0:
+      numel = reduce(mul, tensor.shape)
```
Comment (Thenerdstation) [inline tensornetwork/backends/tensorflow/tensorflow_backend.py:392]: We generally prefer using `np.prod`.

File: tensornetwork/block_sparse/blocksparsetensor.py [hunk 2]
```diff
@@ -430,10 +431,15 @@ def __repr__(self):
     return output
 
   def item(self):
-    if self.ndim == 0:
-      if len(self.data) == 1:
-        return self.data[0]
-    raise ValueError("can only convert an array of size 1 to a Python scalar")
+    if len(self.shape) > 0:
+      if reduce(mul, self.shape) != 1:
```
Comment (Thenerdstation) [inline tensornetwork/block_sparse/blocksparsetensor.py:435]: `np.prod` here too

**Review by Thenerdstation — APPROVED** [review]

File: tensornetwork/backends/tensorflow/tensorflow_backend.py (hunk 1 above)
Comment (mganahl) [reply tensornetwork/backends/tensorflow/tensorflow_backend.py:392]: np.prod has a significantly larger overhead for small sequences, this is why I'm using reduce here

Comment (mganahl) [reply tensornetwork/backends/tensorflow/tensorflow_backend.py:392]: it's not important though

Comment (Thenerdstation) [reply tensornetwork/backends/tensorflow/tensorflow_backend.py:392]: If it's actually faster than we can keep it.

**Comment by Thenerdstation** [comment]
Feel free to merge this martin

File: tensornetwork/backends/tensorflow/tensorflow_backend.py [hunk 3]
```diff
@@ -17,6 +17,8 @@
 from tensornetwork.backends import abstract_backend
 from tensornetwork.backends.tensorflow import decompositions
 from tensornetwork.backends.tensorflow import tensordot2
+from functools import reduce
```
Comment (alewis) [inline tensornetwork/backends/tensorflow/tensorflow_backend.py:20]: Only import modules

**Review by alewis — CHANGES_REQUESTED** [review]

**Comment by alewis** [comment]
As you're probably getting by now I think we should not import classes or functions directly but only modules (which you can give a short name if you want). Otherwise looks good.

**Comment by mganahl** [comment]
taking executive action

**Pull request merged**